### PR TITLE
Allow filtering in or out commits based on strings within commit message

### PR DESCRIPTION
**Pull request by eliotstocker**
It was a requirement to be able to filter in/out release commits for a pipeline we have been working on.
this should allow a lot of customization in the way of skipping commits etc based on commit message

### CODE REVIEW

**Comment by eliotstocker** [comment]
This should also mostly address: https://github.com/concourse/git-resource/issues/215

File: README.md [hunk 1]
```diff
@@ -86,6 +86,12 @@ Tracks the commits in a [git](http://git-scm.com/) repository.
     * `proxy_port`: *Required.* The proxy server's listening port
     * `proxy_user`: *Optional.* If the proxy requires authentication, use this username
     * `proxy_password`: *Optional.* If the proxy requires authenticat, use this password
+    
+* `filter`: *Optional.* Object containing commit message filters
```
Comment (vito) [inline README.md]: Could we rename this to `commit_filter` to be more explicit?

File: assets/check [hunk 2]
```diff
@@ -28,6 +28,9 @@ tag_filter=$(jq -r '.source.tag_filter // ""' < $payload)
 git_config_payload=$(jq -r '.source.git_config // []' < $payload)
 ref=$(jq -r '.version.ref // ""' < $payload)
 skip_ci_disabled=$(jq -r '.source.disable_ci_skip // false' < $payload)
+filterWhitelist=$(jq -r '.source.filter.include // []' < $payload)
+filterBlacklist=$(jq -r '.source.filter.exclude // []' < $payload)
```
Comment (vito) [inline assets/check]: nitpick: consistent variable naming style (`filter_whitelist`) - goes for all the other vars too. :sweat_smile:

**Comment by eliotstocker** [comment]
Changes Applied 😄 

Ubuntu tests are failing for some reason, not sure why they run fine locally

**Comment by jakubikan** [comment]
any update on this? 

I could have a use case on running a pipeline on certain commit messages. 

Any idea when this review will be finished?

**Pull request merged**